### PR TITLE
keg: fix fallback dependencies with multiple kegs

### PR DESCRIPTION
**Pull request by alyssais**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

We didn't remove formulae to be uninstalled from the list of installed formulae to check for dependencies, but only in the fallback. This is what caused #1524, and made it difficult to reproduce (because one had to realise that it was only in the fallback).

### CODE REVIEW

File: Library/Homebrew/keg.rb [hunk 1]
```diff
@@ -105,7 +105,13 @@ def self.find_some_installed_dependents(kegs)
     #
     # This happens after the initial dependency check because it's sloooow.
     remaining_formulae = Formula.installed.select do |f|
-      f.installed_kegs.any? { |k| Tab.for_keg(k).runtime_dependencies.nil? }
+      installed_kegs = f.installed_kegs
+
+      # All installed kegs are going to be removed anyway,
```
Comment (MikeMcQuaid) [inline Library/Homebrew/keg.rb]: Can you elaborate a bit more on this? The "removed anyway" bit makes sense in context of the PR but I'm not sure it does in the context of the function itself which isn't necessarily relating specifically to `uninstall`.

Comment (alyssais) [inline Library/Homebrew/keg.rb]: Sure. The purpose of `Keg.find_some_installed_dependents` is to take some kegs and see if there are any kegs outside of that array that depend on those kegs. (That's why it takes an array.) This is what it [does](https://github.com/Homebrew/brew/blob/c29a458dc15efc92e3649155d0ed44bf982cdc15/Library/Homebrew/keg.rb#L95) currently when the install receipts are reliable.

But, in the fallback, I forgot to remove the passed kegs from the list of dependents, so now I've fixed that.

Comment (MikeMcQuaid) [inline Library/Homebrew/keg.rb]: Cool, makes sense. I think the comment could be expanded on a bit to make this clearer e.g. why they will be "removed"

Comment (alyssais) [inline Library/Homebrew/keg.rb]: Will do 👍

**Comment by MikeMcQuaid** [comment]
Perfect. Great comments.

**Comment by MikeMcQuaid** [comment]
Thanks @alyssais!

**Pull request merged**